src/invocation/ClientConnectionManager.ts

```diff
@@ -0,0 +1,56 @@
+import Q = require("q");
+
+import Address = require("../Address");
+import ClientConnection = require("./ClientConnection");
+
+import OwnerConnectionPicker = require("./OwnerConnectionPicker")
+import InvocationService = require("./InvocationService")
+
+import {GroupConfig} from "../Config";
+
+import ConnectionAuthenticator = require("./ConnectionAuthenticator");
+
+class ClientConnectionManager {
+
+    private groupConfig:GroupConfig;
+    private ownerConnection:ClientConnection;
+    private ownerConnectionPicker:OwnerConnectionPicker;
+    private invocationService:InvocationService;
+
+    private ready = Q.defer<ClientConnectionManager>();
+
+    constructor(addresses:Address[], groupConfig:GroupConfig, invocationService:InvocationService) {
```

Comment:
Addresses will be provided by higher level component from ClientNetworkConfig. There is no point in passing the entire object here